components/suggest/src/store.rs

```diff
@@ -708,6 +716,60 @@ mod tests {
 
     use crate::SuggestionProvider;
 
+    /// Builder for `SuggestionQuery` to avoid breaking existing tests when it's
+    /// modified. See also the query macros below.
+    pub(crate) struct SuggestionQueryBuilder {
+        keyword: String,
+        providers: Vec<SuggestionProvider>,
+        limit: Option<i32>,
+        phantom_suggestion_type: Option<String>,
+    }
+
+    impl SuggestionQueryBuilder {
+        pub fn new(keyword: String, providers: Vec<SuggestionProvider>) -> SuggestionQueryBuilder {
+            Self {
+                keyword,
+                providers,
+                limit: None,
+                phantom_suggestion_type: None,
+            }
+        }
+
+        pub fn limit(&mut self, limit: i32) -> &Self {
+            self.limit = Some(limit);
+            self
+        }
+
+        pub fn phantom_suggestion_type(&mut self, phantom_type: String) -> &Self {
+            self.phantom_suggestion_type = Some(phantom_type);
+            self
+        }
+
+        pub fn build(&self) -> Result<SuggestionQuery> {
+            Ok(SuggestionQuery {
+                keyword: self.keyword.clone(),
+                providers: self.providers.clone(),
+                limit: self.limit,
+                phantom_suggestion_type: self.phantom_suggestion_type.clone(),
+            })
+        }
+    }
+
+    /// Example:
+    /// query_builder!("literal search string", vec![SuggestionProvider::Amp]).limit(1).build()?
+    macro_rules! query_builder {
+        ($keyword_str:literal, $providers_vec:expr $(,)?) => {
+            SuggestionQueryBuilder::new($keyword_str.into(), $providers_vec)
+        };
+    }
+
+    /// Example: query!("literal search string", vec![SuggestionProvider::Amp])
+    macro_rules! query {
+        ($keyword_str:literal, $providers_vec:expr $(,)?) => {
+            query_builder!($keyword_str, $providers_vec).build()?
+        };
+    }
```

Comment:
Could this be a function?  If so, I think I'd rather use that than a macro to keep things simple.